app/components/markdown/markdown_latex_block/index.tsx

```diff
@@ -147,45 +165,28 @@ const LatexCodeBlock = ({content, theme}: Props) => {
                 theme,
             });
         }
-    }, [managedConfig, intl, insets, theme]);
-
-    const splitContent = () => {
-        const lines = splitLatexCodeInLines(content);
-        const numberOfLines = lines.length;
-
-        if (numberOfLines > MAX_LINES) {
-            return {
-                content: lines.slice(0, MAX_LINES),
-                numberOfLines,
-            };
-        }
+    }, [managedConfig.copyAndPasteProtection, intl, insets, theme]);
 
-        return {
-            lines,
-            numberOfLines,
-        };
-    };
-
-    const onErrorMessage = (errorMsg: Error) => {
-        return <Text style={styles.errorText}>{'Error: ' + errorMsg.message}</Text>;
-    };
-
-    const onRenderErrorMessage = ({error}: {error: Error}) => {
+    const onRenderErrorMessage = useCallback(({error}: {error: Error}) => {
         return <Text style={styles.errorText}>{'Render error: ' + error.message}</Text>;
-    };
+    }, []);
 
-    const languageDisplayName = getHighlightLanguageName('latex');
-    const {lines, numberOfLines} = splitContent();
+    useEffect(() => {
+        if (prevContent.current !== content) {
```

Comment:
If prevContent is not equal to current content, this effect would have not triggered (because of the dependencies). If you want to skip the first render, you can always use `useDidUpdate`.

Nevertheless, as I said in a previous comment, we probably want to get rid of the effect altogether and use useMemo.